src/lib/sql/sql_pipeline.hpp

```diff
@@ -48,29 +39,18 @@ class SQLPipeline : public Noncopyable {
               const std::shared_ptr<PreparedStatementCache>& prepared_statements,
               const CleanupTemporaries cleanup_temporaries);
 
-  // Returns the SQL string for each statement.
+  // Vectors containing one element for each SQL statement
   const std::vector<std::string>& get_sql_strings();
-
-  // Returns the parsed SQL string for each statement.
   const std::vector<std::shared_ptr<hsql::SQLParserResult>>& get_parsed_sql_statements();
-
-  // Returns the unoptimized LQP root for each statement.
   const std::vector<std::shared_ptr<AbstractLQPNode>>& get_unoptimized_logical_plans();
-
-  // Returns the optimized LQP root for each statement.
   const std::vector<std::shared_ptr<AbstractLQPNode>>& get_optimized_logical_plans();
-
-  // Returns the SQLQueryPlan for each statement.
-  // For now, this always uses the optimized LQP.
+  // The plans are either retrieved from the SQLPlanCache or, if unavailable, translated from the optimized LQPs
   const std::vector<std::shared_ptr<SQLQueryPlan>>& get_query_plans();
 
-  // Returns all tasks for each statement that need to be executed for this query.
```

Comment:
The return type is everything but self-explanatory. Removing this comment greatly reduces the readability.

Reply:
Ah, sorry, overlooked that this is a vector of vectors.